module/input/PS3Controller/src/PS3Controller.cpp

```diff
@@ -0,0 +1,244 @@
+                            break;
+                    }
+                }
+                else if (is_axis) {
+                    float normalised =
+                        static_cast<float>(value) / static_cast<float>(std::numeric_limits<int16_t>::max());
+                    switch (static_cast<Axis>(event_buffer.front())) {
+                        case Axis::JOYSTICK_LEFT_HORIZONTAL:
+                            emit(std::make_unique<LeftJoystick>(
+                                timestamp, LeftJoystick::Direction::HORIZONTAL, normalised, is_init));
+                            break;
+                        case Axis::JOYSTICK_LEFT_VERTICAL:
+                            emit(std::make_unique<LeftJoystick>(
+                                timestamp, LeftJoystick::Direction::VERTICAL, normalised, is_init));
+                            break;
+                        case Axis::L2: emit(std::make_unique<L2Trigger>(timestamp, normalised, is_init)); break;
+                        case Axis::JOYSTICK_RIGHT_HORIZONTAL:
+                            emit(std::make_unique<RightJoystick>(
+                                timestamp, RightJoystick::Direction::HORIZONTAL, normalised, is_init));
+                            break;
+                        case Axis::JOYSTICK_RIGHT_VERTICAL:
+                            emit(std::make_unique<RightJoystick>(
+                                timestamp, RightJoystick::Direction::VERTICAL, normalised, is_init));
+                            break;
+                        case Axis::R2: emit(std::make_unique<R2Trigger>(timestamp, normalised, is_init)); break;
+                        case Axis::ACCELEROMETER_X: /* TODO: Handle this. */ break;
+                        case Axis::ACCELEROMETER_Y: /* TODO: Handle this. */ break;
+                        case Axis::ACCELEROMETER_Z: /* TODO: Handle this. */ break;
+                    }
+                }
+                else {
+                    log<NUClear::WARN>("Unknown event on joystick. Ignoring.");
+                    log<NUClear::WARN>(fmt::format("Timestamp: {}", timestamp.time_since_epoch().count()));
+                    log<NUClear::WARN>(fmt::format("Value....: {}", value));
+                    log<NUClear::WARN>(
+                        fmt::format("Type.....: {}", is_init ? type | static_cast<uint8_t>(EventType::INIT) : type));
+                    log<NUClear::WARN>(fmt::format("Number...: {}", static_cast<int>(event_buffer.front())));
+                }
+
+                // Erase axis/button from buffer
+                event_buffer.erase(event_buffer.begin(), std::next(event_buffer.begin(), sizeof(uint8_t)));
+            }
+        });
+
+        // Trigger when the joystick accelerometer has an event to read
+        on<IO>(accelerometer_fd, IO::READ).then([this] {
+            // Accelerometer comes in here .... if you know which axis is which
```

Comment:
Should we read this? I think there is also gyroscope information in these packets too

Reply:
I havent been able to figure out the data yet, and we dont use it, so I left it as it is for further use

Reply:
Ahh, the old "exercise for the reader"

Reply:
yep, one of those